azurerm/internal/services/postgres/postgresql_flexible_server_database_resource.go

```diff
@@ -0,0 +1,162 @@
+)
+
+func resourcePostgresqlFlexibleServerDatabase() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourcePostgresqlFlexibleServerDatabaseCreate,
+		Read:   resourcePostgresqlFlexibleServerDatabaseRead,
+		Delete: resourcePostgresqlFlexibleServerDatabaseDelete,
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.FlexibleServerDatabaseID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FlexibleServerDatabaseName,
+			},
+
+			"server_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FlexibleServerID,
+			},
+
+			"charset": {
+				Type:             pluginsdk.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc:     validate.DatabaseCharset,
+			},
+
+			"collation": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.DatabaseCollation,
+			},
+		},
+	}
+}
+
+func resourcePostgresqlFlexibleServerDatabaseCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	client := meta.(*clients.Client).Postgres.FlexibleServerDatabaseClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	serverId, err := parse.FlexibleServerID(d.Get("server_id").(string))
+	if err != nil {
+		return err
+	}
+
+	id := parse.NewFlexibleServerDatabaseID(subscriptionId, serverId.ResourceGroup, serverId.Name, name)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, serverId.ResourceGroup, serverId.Name, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for present of existing %q: %+v", id, err)
+			}
+		}
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return tf.ImportAsExistsError("azurerm_postgresql_flexible_server_database", id.ID())
+		}
+	}
+
+	properties := postgresqlflexibleservers.Database{
+		DatabaseProperties: &postgresqlflexibleservers.DatabaseProperties{
+			Charset:   utils.String(d.Get("charset").(string)),
+			Collation: utils.String(d.Get("collation").(string)),
+		},
+	}
+
+	future, err := client.Create(ctx, id.ResourceGroup, id.FlexibleServerName, id.DatabaseName, properties)
+	if err != nil {
+		return fmt.Errorf("creating Azure Postgresql Flexible Server database %q on server %q (Resource Group %q): %+v", id.DatabaseName, id.FlexibleServerName, id.ResourceGroup, err)
```

Comment:
this can be simplified to:

```suggestion
		return fmt.Errorf("creating %s: %+v", id, err)
```

Reply:
fixed